graphql/index.ts

```diff
@@ -1,10 +1,34 @@
+const mockCachedBooks = [
+  {id: '1', name: 'default book', authors: ['author1', 'author2']}
+]
+
 export const resolvers = {
   Query: {
-    sampleStaticResolver: (_, body) => {
-      console.log(body)
-      return {
-        result: 'static with param: ' + body.param
+    cachedBooks: (_, __) => {
+      return mockCachedBooks
+    },
+    cachedBook: (_, {id}) => {
+      return mockCachedBooks.find((book) => book.id === id)
+    },
+  },
+  Mutation: {
+    deleteCachedBook: (_, {id}) => {
+      const index = mockCachedBooks.findIndex((book) => book.id === id)
+
+      if (-1 < index && index < mockCachedBooks.length) {
+        mockCachedBooks.splice(index, 1)
+        return true
       }
+      return false
+    },
+    createCachedBook: (_, {data: {authors, name}}) => {
+      const book = {
+        authors,
+        id: (mockCachedBooks.length+1).toString(),
```

Comment:
Ids may conflict here because of the way books are being deleted. Best to have a separate counter for id, instead of using the length of the array, which is dynamic.

Reply:
Nice ! I haven't realized it before. I've just refactored it :)